crates/cxx-qt-build/src/lib.rs

```diff
@@ -715,26 +767,104 @@ impl CxxQtBuilder {
         // https://github.com/rust-lang/rust/issues/108081
         // https://github.com/KDAB/cxx-qt/pull/598
         if qtbuild.version().major == 5 {
-            let std_types_contents = include_str!("std_types_qt5.cpp");
-            let std_types_path = format!(
-                "{out_dir}/std_types_qt5.cpp",
-                out_dir = env::var("OUT_DIR").unwrap()
-            );
-            let mut source =
-                File::create(&std_types_path).expect("Could not create std_types source");
-            write!(source, "{std_types_contents}").expect("Could not write std_types source");
-            cc_builder_whole_archive.file(&std_types_path);
-            cc_builder_whole_archive_files_added = true;
+            self.initializers
+                .push(include_str!("std_types_qt5.cpp").to_owned());
         }
+    }
 
-        if cc_builder_whole_archive_files_added {
-            cc_builder_whole_archive.compile("qt-static-initializers");
+    fn build_initializers(&mut self, init_builder: &cc::Build) {
+        let initializers_path = out_dir().join("cxxqt_initializers.cpp");
+        std::fs::write(&initializers_path, self.initializers.join("\n"))
+            .expect("Could not write cxx_qt_initializers.cpp");
+        Self::build_object_file(
+            init_builder,
+            initializers_path,
+            Some(("", "initializers.o")),
+        );
+    }
+
+    fn build_qrc_files(&mut self, init_builder: &cc::Build, qtbuild: &mut qt_build_utils::QtBuild) {
+        for qrc_file in &self.qrc_files {
+            let obj_file_dir = format!("qrc/{}", crate_name());
+            let obj_file_name = format!(
+                "{}.o",
+                qrc_file.file_name().unwrap_or_default().to_string_lossy()
```

Comment:
or_default seems fun here? could it try and create a `qrc/.o` file?

Reply:
Hm, good question, I guess you could...
I'll change it to unwrap_or_else and panic! instead

Reply:
On the other hand, we're currently saying that `.qrc` won't work with CMake anyway, so we also don't really need to export this as an object file. From CMake it's far simpler to just re-add the qrc file as a source.

So I think I'll just not export the object file in that case.

Reply:
I'll also add a warning. to the documentation of `.qrc`

Reply:
For now, but once all the changes are in `qrc` should work via a Rust build script with a CMake build, so we are working towards making it work nicely

Reply:
Yes, absolutely :+1: